addons/udes_stock/models/stock_picking_batch.py

```diff
@@ -510,14 +518,22 @@ def add_extra_pickings(self, picking_type_id):
         """ Get the next available picking and add it to the current users batch """
         Picking = self.env['stock.picking']
 
+        if not self.u_ephemeral:
+            raise ValidationError(_("Can only add work to ephemeral batches"))
+
         picking_priorities = self.get_batch_priority_group()
         pickings = Picking.search_for_pickings(picking_type_id, picking_priorities)
 
         if not pickings:
             raise ValidationError(_("No more work to do."))
 
-        if not self.u_ephemeral:
-            raise ValidationError(_("Can only add work to ephemeral batches"))
+        reserve_pallet_per_picking = self.picking_type_ids.filtered('u_reserve_pallet_per_picking')
+        if reserve_pallet_per_picking:
+            max_reservable_pallets = self.picking_type_ids.mapped('u_max_reservable_pallets')[0]
```

Comment:
If we get the picking type from pickings, then this line does not need `.mapped()` and neither need `[0]`.